### PR TITLE
Domestic component refactor

### PR DESCRIPTION
**Pull request by albrja**
### Refactors domestic components to use vivarium Component
- *Category*: Refactor
- *JIRA issue*: [MIC-4414](https://jira.ihme.washington.edu/browse/MIC-4414)
- *Research reference*: 

### Changes and notes
-refactors components to use vivarium Component

### Verification and Testing
Successfully ran simulation.

### CODE REVIEW

File: src/vivarium_census_prl_synth_pop/components/businesses.py [hunk 1]
```diff
@@ -107,18 +110,6 @@ def setup(self, builder: Builder):
...
-        builder.population.initializes_simulants(
-            self.on_initialize_simulants,
-            requires_columns=["age"],
-            creates_columns=self.columns_created,
-        )
-        # note: priority must be later than that of migration components
-        builder.event.register_listener(
-            "time_step",
-            self.on_time_step,
-            priority=metadata.PRIORITY_MAP["businesses.on_time_step"],
```
Comment (rmudambi) [inline src/vivarium_census_prl_synth_pop/components/businesses.py:119]: I think you forgot to set the priority?

File: src/vivarium_census_prl_synth_pop/components/person_emigration.py [hunk 2]
```diff
@@ -78,16 +78,6 @@ def setup(self, builder: Builder):
...
         )
 
-        builder.population.initializes_simulants(
-            self.on_initialize_simulants,
-            creates_columns=["in_united_states"],
-        )
-        builder.event.register_listener(
-            "time_step",
-            self.on_time_step,
-            priority=metadata.PRIORITY_MAP["person_emigration.on_time_step"],
```
Comment (rmudambi) [inline src/vivarium_census_prl_synth_pop/components/person_emigration.py:88]: I think you forgot to set the priority?

File: src/vivarium_census_prl_synth_pop/components/population.py [hunk 3]
```diff
@@ -60,8 +60,21 @@ def columns_created(self) -> List[str]:
         ]
 
     @property
-    def colums_required(self) -> List[str]:
-        return ["state_id_for_lookup"]
+    def columns_required(self) -> List[str]:
```
Comment (rmudambi) [inline src/vivarium_census_prl_synth_pop/components/population.py:63]: Was this just missed before?

**Review by rmudambi — CHANGES_REQUESTED** [review]

File: src/vivarium_census_prl_synth_pop/components/population.py (hunk 3 above)
Comment (albrja) [reply src/vivarium_census_prl_synth_pop/components/population.py:63]: Yes.

**Review by rmudambi — APPROVED** [review]

**Pull request merged**